### PR TITLE
i2c: AddressMode: add bound Copy

### PR DESCRIPTION
**Pull request by spikespaz**
Since the trait `AddressMode` is `Sealed`, and only implemented on `Copy` types (`u8` and `u16`), there isn't any reason for the trait not to be bound also as `Copy`. This makes multiple-consumption much easier, and does not require calling code to add extra bounds.

### CODE REVIEW

**Comment by jannic** [comment]
We had some discussion about this on Monday in the matrix room: https://libera.irclog.whitequark.org/rust-embedded/2024-10-07#1728335635-1728335961;

Adding `Copy` is, at least technically, a breaking change. Example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=705af2100f5461aa539eaddb2321bfe4

However we didn't find any _realistic_ situation where one would write code that would be affected. 

There was no conclusion if this should be merged (ignoring the breaking change), or deferred for the next major version.

**Comment by spikespaz** [comment]
We agree that this is a breaking change and not a major one. It might fall under the "additive" category. Isn't there a tool that is used to check how many crates would be broken by small changes like this?

**Review by Dirbaio — APPROVED** [review]
LGTM. 

As commented above and discussed in last's week meeting, this is technically breaking but is unlikely to be hit by anyone, since it affects `dyn AddressMode` and there was no way to do anything useful with it anyway. 

I'll wait until meeting on next Tuesday to merge in case another HAL team member wants to raise a concern.

**Review by therealprof — APPROVED** [review]
LGTM, too

**Pull request merged**
